### ISSUE TITLE
[AVM Module Issue]: finops toolkit/finops hub: ResourceGroupName cannot be null

### ISSUE THREAD

**Issue by ryan-baugh** [Azure/bicep-registry-modules#3312]
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Module Name

avm/ptn/finops-toolkit/finops-hub

### Description

We pulled this finops hub pattern from the [Azure Verified Modules pattern](https:/azure.github.io/Azure-Verified-Modules/indexes/bicep/bicep-pattern-modules/). From my understanding, these are Microsoft supported modules. We did modify this pattern a bit to avoid hitting policy violations within our organization. The only piece of the patten that is not working are the deployment scripts. I’ve got the modified patten deploying everything up until the scripts are ran. To me it seems like an issue with something within the scripts, but I could be wrong on that. Below is a screenshot of the resource group that we are creating and deploying to. As you can see, all pieces of the patten are deployed successfully (even the deployment scripts). I believe the issue is coming from when the scripts are being ran. I’ve tried modifying the scripts a bit to add -resourceGroupName triggers to all the azure PowerShell commands where applicable but that didn't resolve the issue either.

```json
{
  "status": "Failed",
  "error": {
    "code": "DeploymentFailed",
    "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-deployment-operations for usage details.",
    "target": "/subscriptions/********-****-****-****-************/resourceGroups/dev-finops-hub/providers/Microsoft.Resources/deployments/5bdfioohquu2o-uploadSettings",
    "details": [
      {
        "code": "ResourceDeploymentFailure",
        "message": "The resource write operation failed to complete successfully, because it reached terminal provisioning state 'failed'.",
        "target": "/subscriptions/********-****-****-****-************/resourceGroups/dev-finops-hub/providers/Microsoft.Resources/deploymentScripts/uploadSettings",
        "details": [
          {
            "code": "DeploymentScriptOperationFailed",
            "message": "'resourceGroupName' cannot be null."
          }
        ]
      }
    ]
  }
}
```

![image](https://github.com/user-attachments/assets/f24b0834-80a6-4674-99a7-f65c72e5c17f)

**Comment by ryan-baugh** [comment]
@sebassem Let me know if you'd like to connect and I can show you the issue.

**Comment by ryan-baugh** [comment]
@sebassem I will work on getting this into one of our pipelines for deployment but just from the quick glance at the template, it seems like this will need to be modified for our environment as it looks like it's completely open to the public. Another item I noticed right off the bat is I'm curious why the deployment scripts were broken out into variables instead of actual scripts.

I'll work on this and update this once we have a successful deployment or if we run into more issues.

**Comment by sebassem** [comment]
> @sebassem I will work on getting this into one of our pipelines for deployment but just from the quick glance at the template, it seems like this will need to be modified for our environment as it looks like it's completely open to the public. Another item I noticed right off the bat is I'm curious why the deployment scripts were broken out into variables instead of actual scripts.
> 
> I'll work on this and update this once we have a successful deployment or if we run into more issues.

Correct, the new version we are working on that will be published on the link I shared has an option to be deployed completely private.

**Comment by ryan-baugh** [comment]
@sebassem Thanks for the info, is there an ETA when that new version will be published with the private option? It might be worth waiting from our side if that will be soonish.

**Comment by sebassem** [comment]
> @sebassem Thanks for the info, is there an ETA when that new version will be published with the private option? It might be worth waiting from our side if that will be soonish.

The code is ready and going through some testing and validation. I expect by the next release.

adding @arthurclares

**Comment by anjalioleksy** [comment]
request to note that this module is not currently supported in the README, and direction to the source. save time and confusion as this should be a MS supported AVM. If its not it shouldn't be here until it is

**Comment by flanakin** [comment]
@ryan-baugh We are adding support for private endpoints in the next release (0.7). Is that all you need or do you need anything else? If so, can you create an issue in the [FinOps toolkit repo](https://aka.ms/ftk)? That way we can triage and plan for it with our other work. We aren't currently maintaining the AVM version due to performance issues with how AVM modules work.

**Comment by arthurclares** [comment]
This module will be deprecated. Please, use the ARM template for deployment: aka.ms/finops/hubs/deploy . All the future requests for support should be sent to https://github.com/microsoft/finops-toolkit/issues
